src/osd/PrimaryLogPG.cc

```diff
@@ -6221,7 +6221,7 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	notify_info_t n;
 	n.timeout = timeout;
 	n.notify_id = osd->get_next_id(get_osdmap_epoch());
-	n.cookie = op.watch.cookie;
+	n.cookie = op.notify.cookie;
```

Comment:
yeah. as we don't set `op.notify.cookie` when preparing an `OP_NOTIFY`.

https://github.com/ceph/ceph/blob/ed8ff905fc5cc71a0a26eb721b9a68ec0ae8084e/src/osdc/Objecter.h#L1027-L1028

`op.watch.cookie` is set for the `OP_WATCH` see 

https://github.com/ceph/ceph/blob/ed8ff905fc5cc71a0a26eb721b9a68ec0ae8084e/src/osdc/Objecter.h#L1017-L1022

the reason that it's been working is just the memory layout of `watch` and `notify` put `cookie` member variable in beginning of the struct.